File: c/pe003.c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <math.h>

int main()
{
  unsigned long long number = 600851475143ull;
  int max_prime = 0;
  int i = 3;
  while (number != 1)
  {
    if ((number % i))
    {
      i+=2;
    }
    else
    {
      max_prime = i;
      number = number/i;
    }
  }
  printf("%d\n", max_prime);
  exit(0);
}

File: go/pe003.go
package main

import (
	"fmt"
)

func main() {
	var number uint64 = 600851475143
	var i uint64 = 3
	var max_prime uint64
	for number != 1 {
		if number%i == 0 {
			max_prime = i
			number = number / i
		} else {
			i = i + 2
		}
	}
	fmt.Printf("%d\n", max_prime)
}
